paasta_tools/autoscaling_lib.py

```diff
@@ -775,19 +778,18 @@ class ClusterAutoscalingError(Exception):
     pass
 
 
-def get_instance_id_from_ip(ip):
+def get_instance_id_from_ip(ip, instance_descriptions):
```

Comment:
Yeah, I can see why it would be nice to have the logic separately. Don't think the `raise_unless` function would help that much because the context changes depending on the thing you're checking. I'll try out your first suggestion and post it up 👍